include/wil/registry_basic.h

```diff
@@ -38,1865 +36,1779 @@
-            template <typename R>
-            ::wil::reg::optional_value<R> try_get_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, DWORD type = ::wil::reg::reg_view_details::get_value_type<R>()) const
+            // type T is the owning type for storing a key: HKEY, ::wil::unique_hkey, ::wil::shared_hkey
+            template <typename T, typename err_policy = ::wil::err_exception_policy>
+            class reg_view_t
             {
-                ::wil::reg::optional_value<R> return_value;
-
-                return_value.inner_error = get_value_internal<R, ::wil::err_returncode_policy>(subkey, value_name, return_value.value, type);
-                if (SUCCEEDED(return_value.inner_error))
+            public:
+                template <typename K>
+                explicit reg_view_t(K&& key) WI_NOEXCEPT :
+                    m_key(::wistd::forward<K>(key))
                 {
-                    return_value.value_assigned = true;
-                    return return_value;
                 }
 
-                if (return_value.inner_error == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND))
+                [[nodiscard]] HKEY get_key() const WI_NOEXCEPT
                 {
-                    return_value.value = {};
-                    return return_value;
+                    return ::wil::reg::details::get_key(m_key);
                 }
 
-                // throw if exception policy
-                err_policy::HResult(HRESULT_FROM_WIN32(return_value.inner_error));
-                return return_value;
-            }
-
-            template <typename R>
-            ::wil::reg::optional_value<R>& try_get_value(_In_opt_ PCWSTR value_name, DWORD type = ::wil::reg::reg_view_details::get_value_type<R>()) const
-            {
-                return try_get_value(nullptr, value_name, type);
-            }
-
-            template <typename R>
-            typename err_policy::result set_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const R& value) const
-            {
-                const auto error = ::RegSetKeyValueW(
-                    ::wil::reg::details::get_key(m_key),
-                    subkey,
-                    value_name,
-                    ::wil::reg::reg_view_details::set_value_type(value),
-                    static_cast<BYTE*>(::wil::reg::reg_view_details::get_buffer(value)),
-                    ::wil::reg::reg_view_details::get_buffer_size(value));
-                return err_policy::HResult(HRESULT_FROM_WIN32(error));
-            }
-
-            template <typename R>
-            typename err_policy::result set_value(_In_opt_ PCWSTR value_name, const R& value) const
-            {
-                return set_value(nullptr, value_name, value);
-            }
+                typename err_policy::result open_key(_In_opt_ PCWSTR subKey, _Out_ HKEY* hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const

```

Comment:
I don't think we should expose unsafe functions like this. We should always return managed types.